projects/igniteui-angular/src/lib/core/styles/components/button/_button-theme.scss

```
@@ -604,28 +604,44 @@
             width: $icon-in-button-size;
             height: $icon-in-button-size;
             font-size: $icon-in-button-size;
-            display: inline-flex;
-            order: -1;
+        }
+
+        igx-icon {
             margin-#{$right}: rem(12px);
         }
+
+        * ~ igx-icon {
```

```suggestion
        * + igx-icon {
```
I'd suggest you use the `+` (next sibling) operator as we'd want to match only siblings that follow the icon.